### ISSUE TITLE
Concurrency issue when calling stopParsing()

### ISSUE THREAD

**Issue by ctytgat** [uniVocity/univocity-parsers#231]
Version 2.5.9

I got a flipping test in my project which seems to be related to the ConcurrentCharInputReader.

What my code does is:

```
        CsvParser parser = new CsvParser(parserSettings);
        parser.beginParsing(file, CHARSET);
        String[] parsedHeaders;
        try {
            parsedHeaders = parser.getContext().parsedHeaders();
        } finally {
            parser.stopParsing();
        }
```

Once in a while the test fails with:

```
com.univocity.parsers.common.TextParsingException: java.io.IOException - Stream closed
Parser Configuration: CsvParserSettings:
	Auto configuration enabled=true
	Autodetect column delimiter=false
	Autodetect quotes=false
	Column reordering enabled=true
	Empty value=null
	Escape unquoted values=false
	Header extraction enabled=true
	Headers=null
	Ignore leading whitespaces=true
	Ignore trailing whitespaces=true
	Input buffer size=1048576
	Input reading on separate thread=true
	Keep escape sequences=false
	Keep quotes=false
	Length of content displayed on error=-1
	Line separator detection enabled=true
	Maximum number of characters per column=4096
	Maximum number of columns=512
	Normalize escaped line separators=true
	Null value=null
	Number of records to read=all
	Processor=none
	Restricting data in exceptions=false
	RowProcessor error handler=null
	Selected fields=none
	Skip bits as whitespace=true
	Skip empty lines=true
	Unescaped quote handling=nullFormat configuration:
	CsvFormat:
		Comment character=#
		Field delimiter=,
		Line separator (normalized)=\n
		Line separator sequence=\n
		Quote character="
		Quote escape character="
		Quote escape escape character=null
Internal state when error was thrown: line=1, column=0, record=0, charIndex=369, headers=[a,b,c]
	at com.univocity.parsers.common.AbstractParser.handleException(AbstractParser.java:368)
	at com.univocity.parsers.common.AbstractParser.stopParsing(AbstractParser.java:431)
	at ...
Caused by: java.io.IOException: Stream closed
	at sun.nio.cs.StreamDecoder.ensureOpen(StreamDecoder.java:46)
	at sun.nio.cs.StreamDecoder.read(StreamDecoder.java:148)
	at java.io.InputStreamReader.read(InputStreamReader.java:184)
	at com.univocity.parsers.common.input.concurrent.CharBucket.fill(CharBucket.java:70)
	at com.univocity.parsers.common.input.concurrent.ConcurrentCharLoader.readBucket(ConcurrentCharLoader.java:71)
	at com.univocity.parsers.common.input.concurrent.ConcurrentCharLoader.run(ConcurrentCharLoader.java:88)
	... 1 common frames omitted
```

**Comment by jbax** [comment]
Thanks for reporting the bug. It's a bit tricky to reproduce but I think I got it fixed. I've released a 2.6.2-SNAPSHOT build with the adjustment. 

Thank you for using our parsers!

**Comment by jbax** [comment]
Version 2.6.2 released. Thanks for reporting the bug.
